[DOC] Changed the buffer size description to be clearer (#2670)
Haivision/srt: docs/API/configuration-guidelines.md

### Setting Receiver Buffer Size

When the `SRTO_RCVBUF` option value is set using the `srt_setsockopt(..)` function,

> the provided size in bytes is internally converted to the corresponding size in packets.
> The size of a cell for a single packet in the buffer is defined by the
> `SRTO_MSS` option, which is 1500 by default.  This value, decreased by 28 in
> case of IPv4 (20 bytes for IPv4 header and 8 bytes for UDP header), gives 1472
> bytes per packet to be allocated. The actual memory occupied by the receiver
> buffer will be a multiplicity of that value. For the default 8192 packets it
> makes then 11776kB (11.5MB).
> 
> Further, in every cell, there is 16 bytes of SRT header and the remaining space
> is for the payload.

Reviewer: ```suggestion
the provided size in bytes is internally converted to the corresponding size in packets.
The size of a cell for a single packet in the buffer is defined by the
`SRTO_MSS` option, which is 1500 by default.  This value, decreased by 28 in
the case of IPv4 (20 bytes for the IPv4 header and 8 bytes for the UDP header), gives 1472
bytes per packet to be allocated. The actual memory occupied by the receiver
buffer will be a multiple of that value. For the default 8192 packets it
will be 11776 kB (11.5 MB).

Note that every cell has 16 bytes for the SRT header. The remaining space
is for the payload.
```

As already mentioned, the maximum allowed size of the receiver buffer is limited by the value of `SRTO_FC`.

> The following function returns the buffer size in packets:

Reviewer 2: The following function returns the configured buffer size in packets depending on the `SRTO_RCVBUF`, `SRTO_MSS` and `SRTO_FC` values set:

Author: You meant:

```suggestion
The following function returns the configured buffer size in packets depending on the SRTO_RCVBUF, SRTO_MSS and SRTO_FC values set:
```

```c++
int getRbufSizePkts(int SRTO_RCVBUF, int SRTO_MSS, int SRTO_FC)
{
    // UDP/IPv4 header size is assumed to be 28 bytes
    // 20 bytes IPv4 + 8 bytes of UDP

>     const int UDPHDR_SIZE = 28;
>     const int pkts = (rbuf_size / (SRTO_MSS - UDPHDR_SIZE));

Reviewer 2: ```suggestion
    const int UDP_IPv4_HDR= 28;
    const int pkts = (rbuf_size / (SRTO_MSS - UDP_IPv4_HDR));
```

If the whole remainder of the MTU is expected to be used, payload size is calculated as follows: 

> `bytePayloadSize = MSS - UDP_HDR - SRT_HDR`

Author: ```suggestion
`bytePayloadSize = MSS - UDP_IPv4_HDR - SRT_HDR`
```

where

- `MSS`: Maximum Segment Size (size of the MTU); see `SRTO_MSS` (default: 1500)

> - `UDP_HDR`: 20 bytes for IPv4 + 8 bytes for UDP

Reviewer 2: ```suggestion
- `UDP_IPv4_HDR`: 20 bytes for IPv4 + 8 bytes for UDP
```
